### ISSUE TITLE
Packaging for Debian or Ubuntu

### ISSUE THREAD

**Issue by EleanorEllis** [bleskodev/rubyripper#607]
Would it be possible to package this for Debian or Ubuntu and include it in a PPA or even the Debian or Ubuntu repositories?

**Comment by bleskodev** [comment]
Hi,
yes, this would be a nice feature. However, I have 0 experience with Debian/Ubuntu packaging and even less with the procedure to include the package to a PPA or official repositories.
If someone with more experience can jump in and do it, it would be very appreciated.

**Comment by JBBgameich** [comment]
I updated the packaging from deb-multimedia to work with the current version of rubyripper and debian. The copyright file is probably really out of date, but the packaging works (tested on debian unstable)
https://github.com/JBBgameich/rubyripper-packaging
Building can be done using
```
uscan --repack --compression xz --download-current-version
```
(Then unpack the newly downloaded tar.xz file into the packaging directory)
```
apt build-dep .
dpkg-buildpackage
```

**Comment by bleskodev** [comment]
Hey, nice work, thanks a lot for this!
Do I need to take any action?
I don't see any Debian packaging related files in the original rubyripper repository, so I suppose it was done elsewhere as you did it.

**Comment by JBBgameich** [comment]
The uscan command I posted above will download the original source so you can build the deb package from it. You don't necessarily have to add source and packaging to the same repository, but you can.

**Comment by nafanz** [comment]
@bleskodev perhaps worth looking at creating a snap package?

They are now supported in most distros. Unfortunately, I am not a developer and I can not provide information on their creation. As far as I own information from the Internet, this makes it possible to create an isolated container with an application that

Perhaps you will find the website useful: https://snapcraft.io

Perhaps this will also help make the application more popular and thereby attract new contribution ;-)

**Comment by bleskodev** [comment]
@nafanz  You are right. This is maybe not a bad idea. I'll investigate to see what needs to be done. If nothing else, I'll learn something new!

**Comment by Thovthe** [comment]
@bleskodev 
@refi64 Has submitted a pull request for packaging as a Flatpak which may address this need.

**Comment by FoilBlue** [comment]
sudo apt  install libdiscid0 ruby-gettext ruby-gtk3 cd-discid libcdio-utils  setcd dvd+rw-tools cd-discid libdsk-utils
was what i used to install deps in 2021 in ubuntu. there may be a pkgs not needed. as i added a few in case they had the dep in them. turned out it was cd-discid libdsk-utils that had those two tricky to find deps. no luck finding wavegain.

**Comment by poddmo** [comment]
These are the steps I took to install on Ubuntu 2204:
`
sudo apt install cdparanoia ruby ruby-rexml ruby-gettext ruby-gtk3 cd-discid libcdio-utils eject flac flactag vorbis-tools vorbisgain lame mp3gain wavpack replaygain loudgain normalize-audio sox cdrdao

sudo ln -s /usr/bin/wvgain /usr/local/bin/wavegain

git clone https://github.com/bleskodev/rubyripper.git
cd rubyripper 
sudo ./configure --enable-gtk3 --enable-cli
sudo make install
`
